website/content/contribute/module/index.md

```diff
@@ -0,0 +1,199 @@
+	// +optional
+	Read []ReadModuleArgs `json:"read,omitempty"`
+
+	// WriteArgs are parameters that are specific to modules that enable an application to write data
+	// +optional
+	Write []WriteModuleArgs `json:"write,omitempty"`
+}
+```
+
+The status of the module workload is monitored by the control plane via standard kubernetes mechanisms.  
+TODO: Expand on this or delete it?
+
+If the module workload needs to return information to the user, that information should be written to the rel.Info.Note. 
+TODO: Where should it be written to??
+
+## Module Helm Chart
+For any module chosen by the control plane to be part of the data path, the control plane needs to be able to install/remove/upgrade an instance of the module. Mesh for Data uses [Helm](https://helm.sh/docs/intro/using_helm/)
+to provide this functionality.  Thus, a [helm chart](https://helm.sh/docs/chart_template_guide/getting_started/) should be provided for each `Module Workload`.
+
+Example: [Arrow Flight Module helm chart](https://github.com/IBM/the-mesh-for-data-flight-module/tree/cd168bb6cdf666c2ec1df960395c0dc1c8feeaa9/helm/afm)
+
+
+## M4DModule YAML
+M4DModule is a kubernetes Custom Resource Definition (CRD) which describes to the control plane the functionality provided by the module.  The M4DModule CRD has no controller. The specification of the `M4DModule` Kubernetes CRD is available in the [API documentation]({{< baseurl >}}/docs/reference/api/generated/app/#k8s-api-github-com-ibm-the-mesh-for-data-manager-apis-app-v1alpha1-m4dmodule). 
+
+### Metadata about the module
+The first section in the YAML file contains general information about the module and where it will run.
+
+```
+apiVersion: app.m4d.ibm.com/v1alpha1 # always this value
+kind: M4DModule # always this value
+metadata:
+  name: "<module name>" # the name of your new module
+  namespace: m4d-system  # control plane namespace. Always m4d-system
+```
+
+### spec.chart
+This is a link to a the Helm chart stored in an OCI image registry. This is similar to how a Kubernetes `Pod` references a container image. See [Module Helm chart](#module-helm-chart) for more details.
+
+```
+spec:
+  chart: "<helm chart link>" # e.g.: ghcr.io/username/chartname:chartversion
+```
+
+### statusIndicators
+Used for tracking the status of the module in terms of success or failure. In many cases this can be omitted and the status will be detected automatically.
+
+TODO: In which cases is it automatically detected and when is it not?
+```
+statusIndicators:
+    - kind: <module name>
+      successCondition: status.status == SUCCEEDED
+      failureCondition: status.status == FAILED
+      errorMessage: status.error
+```
+
+
+### dependencies
+A dependency has a `type` and a `name`. Currently dependencies of type `module` are supported, indicating that another module must also be installed for this module to work.
```

Comment:
**only** dependencies of type 'module' are supported

Reply:
Yes, that is stated